compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -238,73 +241,64 @@ struct TokenCursor {
 struct TokenCursorFrame {
     delim: token::DelimToken,
     span: DelimSpan,
-    open_delim: bool,
     tree_cursor: tokenstream::Cursor,
-    close_delim: bool,
+    need_to_produce_close_delim: bool,
 }
 
 impl TokenCursorFrame {
     fn new(span: DelimSpan, delim: DelimToken, tts: TokenStream) -> Self {
         TokenCursorFrame {
             delim,
             span,
-            open_delim: false,
             tree_cursor: tts.into_trees(),
-            close_delim: false,
+            need_to_produce_close_delim: delim != DelimToken::NoDelim,
```

Comment:
```suggestion
            need_to_produce_close_delim: true,
```
Can all the `NoDelim` treatment for the closing token be moved to `fn inlined_next`, in a similar way to the opening token?
Then it would be the place to add the FIXME about https://github.com/rust-lang/rust/issues/67062.

Reply:
Ok, I will do this.